pep-9999.rst

```diff
@@ -0,0 +1,520 @@
+matrices (just as they test against the latest maintenance release of previously
+published versions), with the CPython nightly builds offered by some CI
+providers used solely in an advisory capacity for early detection of potential
+compatibility problems.
+
+
+Implications for the proposed Scientific Python ecosystem support period
+------------------------------------------------------------------------
+
+Based on discussions at SciPy 2019, a NEP (NumPy Enhancement Proposal) is
+currently being drafted [3_] to define a common convention across the
+Scientific Python ecosystem for dropping support for older Python versions.
+
+While the exact formulation of that policy is still being discussed, the initial
+proposal was very simple: support any Python feature release published within
+the last 42 months.
+
+For an 18 month feature release cadence, that works out to always supporting at
+least the two most recent feature releases, and then dropping support for all
+X.Y.z releases around 6 months after X.(Y+2).0 is released. This means there is
+a 6 month period roughly every other year where the three most recent feature
+releases are supported.
+
+For a 12 month release cadence, it would work out to always supporting at
+least the three most recent feature releases, and then dropping support for all
+X.Y.z releases around 6 months after X.(Y+3).0 is released. This means that
+for half of each year, the four most recent feature releases would be supported.
+
+For a 24 month release cadence, a 42 month support cycle works out to always
+supporting at least the most recent feature release, and then dropping support
+for all X.Y.z feature releases around 18 months after X.(Y+1).0 is released.
+This means there is a 6 month period every other year where only one feature
+release is supported. If the support cycle is increased to 48 months, then
+that restores the characteristics of always supporting at least the two most
+recent feature releases.
+
+
+Release cycle alignment for core development sprints
+----------------------------------------------------
+
+With the proposal in this PEP, it is expected that the focus of core
+development sprints will shift focus slightly based on the current location
+in the two year cycle.
+
+In odd-numbered years, PyCon US would naturally focus on release candidate
+preparations, while the September core development sprints would focus on
+closing out any final non-security bug fixes for the release series entering
+its security-fix-only phase, the initial set of post-release fixes for the just
+released series, and preparing for the first beta release in the next
+release series.
+
+In even-numbered years, the focus for both events would just be on the upcoming
+maintenance and beta releases. These less intense years would hopefully provide
+an opportunity to tackle various process changes and infrastructure upgrades
+without impacting the release candidate preparation process.
```

Comment:
I think you can sell these better :)

PyCon US is a great for greenfield discussions, particularly with those outside the core team. That doesn't lend itself to RC preparations at all, but for those who are not focused on RC can be thinking about big features for the next release.

Similarly, the core sprints are good for process and infrastructure (as mentioned). There'll also be a steering council election around that time every two years.

So I think we can justify this as:
* in non-release years, feature development and maintenance is good for both
* in release years, PyCon US is good for new contributors to work on bugfixes before RC and core devs to plan the next release; core sprints are good for reviewing our process

Reply:
Reworded more along these lines.